float32[]   x_point
float32[]   y_point
int32   state
